modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/raft/MetaStorageWriteHandler.java

```diff
@@ -227,6 +228,17 @@ private void handleWriteWithTime(CommandClosure<WriteCommand> clo, MetaStorageWr
             evictIdempotentCommandsCache(cmd.evictionTimestamp(), context);
 
             clo.result(null);
+        } else if (command instanceof CompactionCommand) {
+            CompactionCommand cmd = (CompactionCommand) command;
+
+            long compactionRevision = cmd.compactionRevision();
+
+            storage.saveCompactionRevision(compactionRevision, context);
+            storage.startCompaction(compactionRevision);
```

Comment:
Why do we have two separate actions? There should only be one

Reply:
Yes, because after deploying the watches we will need to run a background compaction for the latest compaction revision and we will not have the `KeyValueUpdateContext` on hand.

On the other hand, we can add a `startCompactionAfterRecovery` that will do the same thing but without `saveCompactionRevision` and add it `startCompaction`, what do you think?